WhisperAPI/WhisperAPI/Services/SelectSuggestion/SelectSuggestionService.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Linq;
+using WhisperAPI.Models;
+
+namespace WhisperAPI.Services.SelectSuggestion
+{
+    public class SelectSuggestionService : ISelectSuggestionService
+    {
+        public bool UpdateContextWithSelectedSuggestion(ConversationContext conversationContext, SearchQuery searchQuery)
+        {
+            Guid suggestionId = new Guid(searchQuery.Query);
+            SuggestedDocument suggestedDocument = conversationContext.SuggestedDocuments.ToList().Find(x => x.Id == suggestionId);
+            if (suggestedDocument == null)
```

Comment:
Avoid nested code like that. If you switch the condition to != instead, you can have the two ifs on the same level, don't forget that return statements stops the control flow right away and exits the method, you can take advantage of that:

// find in suggestions
if (... != null)
{
	// ADd to suggested documents list
	return true;
}

// find in questions list
if (.. != null)
{
	// ADd in selected question list
	return true;
}

return false;